### ISSUE TITLE
Add a curses-based TUI for interactive alias generation

### ISSUE THREAD

**Issue by ariel-frischer** [ariel-frischer/alias-gen#22]
## Summary

Implement a TUI (Text-based User Interface) in Python using the **curses** library to allow users to interactively select and edit command aliases. Rather than answering prompts in a purely linear, CLI-based fashion, users can scroll through a list of frequently used commands, select which ones to alias, and optionally edit suggested aliases—all within a single interface.

## Motivation

- **Better UX**: A TUI will give users a more visual approach to generating aliases, making it easier to choose or edit suggested aliases on the fly.  
- **Faster Editing**: Instead of re-running the script multiple times or jumping back and forth between prompts, everything can happen in one place.  
- **Reduced Errors**: Presenting commands and aliases in a structured format reduces the chances of typos or missing commands.

## Proposed Features

1. **Curses-based Main Screen**  
   - Display a list of commands along with their frequency and suggested aliases.  
   - Users can navigate using the arrow keys and toggle selection with the spacebar.

2. **Inline Alias Editing**  
   - When a user presses “E” (or enters a specific edit mode) on a highlighted command, open a small input box for alias editing.  
   - Validate the new alias to avoid collisions with built-ins or existing aliases.

3. **Scrolling / Pagination**  
   - Support scrolling through large lists of commands. If the user has hundreds of unique commands, they should be able to paginate or scroll smoothly.

4. **Filter / Search** (Stretch Goal)  
   - Implement a basic “search” mode to filter commands by substring (e.g., typing `/kube` to only see commands containing “kube”).

5. **Confirmation & Save**  
   - After the user selects and optionally edits aliases, they can press a key (e.g., “S”) to confirm.  
   - The script then writes the aliases to the appropriate shell config file (e.g., `~/.bashrc`, `~/.zshrc`, etc.) or displays them with `--stdout` if specified.

## Implementation Outline

1. **History Parsing / Alias Generation**  
   - Reuse the existing logic to parse shell history and generate suggested aliases.  
   - Pass these results (command, frequency, suggested_alias) to the TUI.

2. **Curses Setup**  
   - Create a Python module `tui_curses.py` (or similar).  
   - Use `curses.wrapper(main_screen)` to initialize curses in a `main_screen` function.  
   - Disable the cursor if desired (`curses.curs_set(0)`) to clean up the UI.

3. **UI Layout**  
   - A main window or “pad” that shows a scrollable list of commands.  
   - Keyboard controls:
     - **Up/Down**: Move selection.  
     - **Space**: Toggle selection of the current command.  
     - **E**: Edit the suggested alias of the highlighted command.  
     - **S**: Save and confirm aliases.  
     - **Q**: Quit without saving.

4. **Alias Editing**  
   - When the user presses “E” on a highlighted command:
     - Open a sub-window (or “popup”) to read user input for the alias.  
     - Handle special keys (Esc to cancel, Enter to confirm).  
     - Validate or check for conflicts.

5. **File Write / Confirmation**  
   - On pressing “S,” prompt the user to confirm writing changes to their shell config or to exit/cancel.  
   - If confirmed, proceed with the existing logic of appending or modifying the rc file.  
   - Consider creating a backup of the rc file before changes.

6. **Testing**  
   - Ensure that the TUI remains stable with small and large sets of commands.  
   - Validate that scrolling, pagination, and alias editing function correctly.

## Acceptance Criteria

- [ ] A user can run the script with something like `alias-gen --tui` (name TBD).  
- [ ] The TUI starts with a list of commands sorted by frequency.  
- [ ] The user can scroll through the list, select commands, and edit suggested aliases.  
- [ ] Saving changes writes or appends to the correct shell config (or displays them via `--stdout`).  
- [ ] Edge cases handled (e.g., no commands to alias, extremely long command histories, etc.).  
- [ ] Minimal dependencies (only built-in `curses`).

## Additional Notes

- Initially, we can keep it simple: no advanced color usage or mouse support. Later, we can extend the TUI to include color highlighting or additional shortcuts.  
- This feature should not break existing workflows (like `alias-gen --stdout`). We can isolate the TUI behind a separate flag or command entry point.  

**Let’s discuss any potential design pitfalls or additional functionality before starting on the implementation.**

**Comment by ariel-frischer** [comment]
@devloai resolve this issue
